src/tests/test_pkcs11_high_level.cpp

```diff
@@ -93,14 +93,21 @@ class TestSession
 Test::Result test_module_ctor()
    {
    Test::Result result("Module ctor");
+   
+   try
```

Comment:
I understand the motivation for this but I think it is possible to do in a simpler way that involves less boilerplate. Change `run_pkcs11_tests` to take `std::vector<std::pair<std::string, std::function<Test::Result()>>>& fns` where fns[i].first is the name and fns[i].second in the function to invoke. Then the try/catch that is already in `run_pkcs11_tests` can include the function name as well as the test name in the event of any exceptions.

Reply:
@randombit I implemented your suggestion, thanks for the hint! What do you think about it?
P.S.: I will be offline until Monday. Have a nice weekend! :-)